content/blog/using-polymorphic-json-adapter-on-android.md

```diff
@@ -93,27 +91,28 @@ class Engineer(
 }
 ```
 
-Now, if the app receives a request where the `occupation` is Doctor or Engineer, we expect Moshi to create an instance of `Doctor` or `Engineer` class. However, Moshi does not know which property it has to use to determine the type of the derived class. Let's solve this.
+Now, if you try to parse a JSON body where the `occupation` is Doctor or Engineer, we want Moshi to create an instance of `Doctor` or `Engineer` class respectively. However, Moshi cannot automatically determine this. Let's help it along.
 
-### Here's how you can do it
+### Creating a polymorphic adapter
 
-1. First, you'll have to add the `moshi-adapters` dependency.
+1. First, add the `moshi-adapters` dependency.
 
    ```kotlin
-   implementation("com.squareup.moshi:moshi-adapters:1.11.0")
+   implementation("com.squareup.moshi:moshi-adapters:<latest version>")
    ```
+
 2. Now create an adapter for your abstract class (`Person` class in this example)
 
    ```kotlin
    val adapterFactory = PolymorphicJsonAdapterFactory
-   						.of(Person::class.java, "occupation")
-   						.withSubtype(Doctor::class.java, "Doctor")
-   						.withSubtype(Engineer::class.java, "Engineer")
+        .of(Person::class.java, "occupation")
+        .withSubtype(Doctor::class.java, "Doctor")
+        .withSubtype(Engineer::class.java, "Engineer")
    ```
 
-   Here's the main part of our Polymorphic adapter. We create an instance of `PolymorphicJsonAdapterFactory` of type `Person` and then we specify which key Moshi should use to determine the derived class it should generate (`occupation` is used here).
+   Here's the main part of our polymorphic adapter. We create an instance of `PolymorphicJsonAdapterFactory` for the type `Person`, and then we specify which key Moshi should use to determine the class it should instantiate. In our case, that's `occupation`.
 
-   After that, we've added all the classes that extend the Person class and specified the values on which those classes should be built. e.g. In this case a JSON like 
+   After that, we add all the subclasses of `Person` and specify the values of `occupation` on which those classes should be selected. In this case, a JSON body like this:
```

Comment:
Thanks for rewording this, I felt mine was terrible but couldn't think of anything else